lib/sneakers/publisher.rb

```diff
@@ -11,7 +11,16 @@ def publish(msg, options = {})
       end
       to_queue = options.delete(:to_queue)
       options[:routing_key] ||= to_queue
-      Sneakers.logger.info {"publishing <#{msg}> to [#{options[:routing_key]}]"}
+      begin
+        # exceptions rescued here because msg is not guaranteed to be JSON
```

Comment:
What can `msg` be besides JSON?

Reply:
`StandardError`

Reply:
in that case what about something like
```rb
      if msg.instance_of? StandardError
        Sneakers.logger.info {"publishing <#{msg}> to [#{options[:routing_key]}]"}
      else
        Sneakers.logger.info {{
          message: "publishing <#{msg}> to [#{options[:routing_key]}]",
          options: options,
          msg: Oj.load(msg)
        }}
      end
```
?

Relying on exception handling for control flow feels unnecessary if we know the message will either be valid JSON or an instance of `StandardError`

Reply:
Oh, it can be a number of other things, including `StandardError`. Apologies for being unclear. I'm going off of how `msg` is handled in the specs, where there's a large amount of branching conditional logic for it being any number of things. I only want to extrapolate JSON in the cases where that's what we have, and return a standard logging message in all other cases.

If there's some way to make OJ always return a string, even when it's not parseable as json, I'd do that.

Reply:
Depends what you're looking for — I believe there are configuration options for Oj that can either:
- Coerce all input into a stringified JSON object
- Fall back to `.to_s` behavior